core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1125,24 +1130,31 @@ class DAGScheduler(
       event.taskInfo.attemptNumber, // this is a task attempt number
       event.reason)
 
-    // The success case is dealt with separately below, since we need to compute accumulator
-    // updates before posting.
```

Comment:
by the way this comment isn't even true. I think this whole "post only failed tasks of cancelled stages" behavior is not intentional. I tried to look into this but there are 30 layers of indirection so I just gave up. Let's tackle this separately (if it actually matters).